### ISSUE TITLE
[frameit] The documentation should show all parameters from the Framefile

### ISSUE THREAD

**Issue by funnel20** [fastlane/fastlane#11647]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
While finding a solution for issue #11585 we found several Framefile parameters in the code, that are not defined in the Frameit documentation at https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/docs/frame_screenshots.md.

From https://github.com/fastlane/fastlane/issues/11585#issuecomment-358693332 and https://github.com/fastlane/fastlane/issues/11585#issuecomment-358694644 these are:

- `padding`
- `font_scale_factor`
- `offset`
- `interline_spacing`

These parameters should be added.

##### Complete output when running fastlane, including the stack trace and command used

[INSERT OUTPUT HERE]

### Environment

[INSERT OUTPUT HERE]

**Comment by funnel20** [comment]
@janpio Can you help with this?

**Comment by janpio** [comment]
Of course - I prompted you to create this issue :)

I would suggest we tackle the whole documentation page you linked to. Historically `frameit` only did framing, then later background, text etc was added - which shows in the way it is documented.

https://docs.fastlane.tools/actions/frame_screenshots/#results is the first time we see more than a simple framed screenshot, but text above it. All the text before didn't even mention that. 

Then after these example screenshots we again only get text that describes the "old" features, until we get to https://docs.fastlane.tools/actions/frame_screenshots/#titles-and-background-optional

That starts with a complete (but confusing) sample configuration, without even introducing (and defining) the functionality. It then lists some configuration options (minus the missing ones we found), but also doesn't really define which ones should be use how etc.

So my idea would be to fix all of these problems:

1. Introduction should mention "base" usage (frame screenshots) but also advanced features (put text around these frames).
2. The advanced features should be described and defined before listing all the options, giving examples etc.

What do you think @funnel20?

**Comment by funnel20** [comment]
This sounds like an excellent approach.

**Comment by janpio** [comment]
If we want to work on it together, taking turns, it would be best if you could create a branch in your fork where you make a tiny edit in that file, and then open a PR here that you mark with `[WIP]` in the title. Then we both can commit to that branch and discuss changes until we are happy and the PR can get merged.

**Comment by janpio** [comment]
Yep, just create a new branch in your fork. It should appear automatically on fastlane/fastlane or when open a new PR you can select it.

**Comment by funnel20** [comment]
Unfortunately I didn't work in a branch for PR #11624, so when making a new PR for the docu all 6 non-relavant commits from the Editor are made.
Sorry, we have to wait until PR #11624 is merged.

**Comment by janpio** [comment]
No, that is normal that the old changes are shown until the previous PR is merged.
(Unless you branch from before you made the changes that are not yet merged)

**Comment by funnel20** [comment]
Ah, maybe that's the case. I'll just wait to verify that.

**Comment by janpio** [comment]
Just for reference, here is another implementation of the same problem: Setting the frames on a background with some text that has some additional features: https://github.com/spreaker/fastlane-framer-plugin

**Comment by funnel20** [comment]
Yes, I mastered multiple brands simultaneously...

**Comment by joshdholtz** [comment]
@funnel20 Just wanted to checkin to see if you are still working on this issue 😊

**Comment by joshdholtz** [comment]
@funnel20 Would you like me to review it?

**Comment by funnel20** [comment]
@joshdholtz Any progress is good I assume, but I don't won't to bypass @janpio.
But feel to free to review my changes and comment.
The biggest change is that instead of describing an (incomplete) set of parameters in text, I added a few tables to describe the parameters. And that in relation to the JSON layout.
